src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,168 @@
+"""
+Extraction algorithms to compute the statistics from a chunk of images
+"""
+
+__all__ = [
+    "StatisticsExtractor",
+    "PlainExtractor",
+    "SigmaClippingExtractor",
+]
+
+from abc import abstractmethod
+from collections import defaultdict
+
+import numpy as np
+from astropy.stats import sigma_clipped_stats
+from astropy.table import Table
+
+from ctapipe.containers import StatisticsContainer
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import Int
+
+
+class StatisticsExtractor(TelescopeComponent):
+    """
+    Base component to handle the extraction of the statistics from a table
+    containing e.g. charges, peak times and/or charge variances (images).
```

Comment:
I think a better docstring is needed here. It's very unclear what the output is.  Is it a single set of statistics, or a series of statistics aggreggated over (potentially overlapping) time bins?  It's more clear later in the `__call__` docstring, but could be better here. Also this is **pixel-wise** statistics, which is maybe important to mention.